### PR TITLE
EXE: GUI: Add new GUI system.

### PR DESCRIPTION
**Pull request by Warepire**
Changed the GUI system. This is the first step in order to step away from the horribly annoying res-system Microsoft seems to love.
This implementation uses MENUEX and DLGTEMPLATEEX as a core to implement Menu and Dialog classes that a window class may include.

This allows the creation of C++11 class-style implementations of dialogs where class-members can be std:bind bound as DLGPROC callbacks.

This is part 1 of a code refactor to bring the executable up to coding standard and improve the code readability.

### CODE REVIEW

**Comment by Warepire** [comment]
Will remove the pragma warning disables, and update the USAGE comment block for DlgBase to properly refer to the types before merge.

**Comment by Warepire** [comment]
I left more stuff behind than I thought, cleaned up.

File: wintaser/GUI/DlgBase.cpp [hunk 1]
```diff
@@ -0,0 +1,440 @@
...
+    iterator += sizeof(short);
+    *reinterpret_cast<short*>(&(window[iterator])) = h;
+    iterator += sizeof(short);
+
+    /*
+     * Leave menu and windowClass alone
+     */
+    iterator += (sizeof(WORD) * 2);
+
+    if(caption.empty() == false)
```
Comment (Scepheo) [inline wintaser/GUI/DlgBase.cpp]: Space between if and parentheses.

File: wintaser/GUI/Menu.cpp [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+ * {
+ *     DWORD dwType;
+ *     DWORD dwState;
+ *     DWORD menuId;
+ *     WORD bResInfo;
+ *     WCHAR szText[];  <-- Variable length NULL terminated WCHAR array
+ * };
+ */
+
+namespace {
```
Comment (Scepheo) [inline wintaser/GUI/Menu.cpp]: Bracket on new line.

File: wintaser/GUI/Menu.cpp [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+{
+    if (loaded_menu != nullptr)
+    {
+        DestroyMenu(loaded_menu);
+    }
+}
+
+void Menu::AddMenuCategory(std::wstring name, DWORD id, bool enabled, bool last)
+{
+    DWORD state = ((enabled) ? MFS_ENABLED : MFS_DISABLED);
```
Comment (Scepheo) [inline wintaser/GUI/Menu.cpp]: Why the parentheses around "enabled"? Same goes for similar cases in rest of file.

File: wintaser/GUI/Menu.cpp [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+    *reinterpret_cast<DWORD*>(&(menu[iterator])) = type;
+    iterator += sizeof(DWORD);
+    *reinterpret_cast<DWORD*>(&(menu[iterator])) = state;
+    iterator += sizeof(DWORD);
+    *reinterpret_cast<DWORD*>(&(menu[iterator])) = id;
+    iterator += sizeof(DWORD);
+    *reinterpret_cast<WORD*>(&(menu[iterator])) = res;
+    iterator += sizeof(WORD);
+
+    if(type != MFT_SEPARATOR)
```
Comment (Scepheo) [inline wintaser/GUI/Menu.cpp]: Space between if and parentheses.

File: wintaser/GUI/Menu.cpp [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+
+/*
+ * TODO: Make sure attaching a menu fixes the size of the dialog
+ * -- Warepire
+ */
+
+bool Menu::AttachMenu(HWND window)
+{
+    loaded_menu = LoadMenuIndirectW(reinterpret_cast<MENUTEMPLATEW*>(menu.data()));
+    if(loaded_menu == nullptr)
```
Comment (Scepheo) [inline wintaser/GUI/Menu.cpp]: Space between if and parentheses.

File: wintaser/GUI/Menu.cpp [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+    {
+        return false;
+    }
+    if(SetMenu(window, loaded_menu) == FALSE)
```
Comment (Scepheo) [inline wintaser/GUI/Menu.cpp]: Space between if and parentheses. Also, why "== FALSE" instead of "!"?

Comment (Warepire) [inline wintaser/GUI/Menu.cpp]: Clarity, personally I often miss the `!` characters.

**Pull request merged**